File: gcode/src/main/java/org/qw3rtrun/p3d/g/decoder/UnknownStringDecoder.java
package org.qw3rtrun.p3d.g.decoder;

import org.qw3rtrun.p3d.g.event.ReceivedUnknownEvent;
import org.qw3rtrun.p3d.core.msg.UnknownEvent;

import java.util.Optional;

public class UnknownStringDecoder implements GEventDecoder<UnknownEvent> {

    @Override
    public Optional<UnknownEvent> decode(String line) {
        return Optional.of(new ReceivedUnknownEvent(line));
    }
}

File: gcode/src/main/kotlin/org/qw3rtrun/p3d/g/decoder/UnknownStringDecoder.kt
package org.qw3rtrun.p3d.g.decoder

import org.qw3rtrun.p3d.core.msg.UnknownEvent
import org.qw3rtrun.p3d.g.event.ReceivedUnknownEvent
import java.util.Optional

class UnknownStringDecoder : GEventDecoder<UnknownEvent> {

    override fun decode(line: String): Optional<UnknownEvent> {
        return Optional.of(ReceivedUnknownEvent(line))
    }
}
